docs/developer/design-principles.md

```diff
@@ -0,0 +1,52 @@
+# Design Principles
+
+The aim of NGINX Kubernetes Gateway is to become a fundamental infrastructure component within a Kubernetes cluster,
+serving as both an ingress and egress point for traffic directed towards the services (applications) running either
+within or outside the cluster. Leveraging NGINX as a data plane technology, it harnesses the well-established reputation
+of NGINX as an open-source project widely recognized for its role as a web server, proxy, load balancer, and content
+cache. NGINX is renowned for its stability, high performance, security, and rich feature set, positioning it as a
+critical infrastructure tool. Notably, once properly configured and operational, NGINX requires minimal attention,
+making it reliable and "boring" software.
```

Comment:
I would consider removing boring and perhaps replacing it with another adjective, such as "flexible" - to emphasise the paragraph's example of uses cases.

Boring is generally a negative adjective and although many experienced engineers enjoy boring (and might refer to the "Choose Boring Technology" talk, a novice user or one less versed popular ideas & talks might be put off.